### DISCUSSION TITLE
Outdated documentation and lack of resources for beginners on the platform.

### DISCUSSION THREAD

**Discussion by juppdes** [serverpod/serverpod#1926] [General]
Is there any plan to update the documentation, examples and tutorials?
All material I find refers to previous versions of the platform.
Simple email authentication becomes complicated without updated documentation and examples.
I really want to participate, collaborate and promote Serverpod, however, for someone who has just started, the deficiencies in support material, and even the documentation pointing to outdated materials, discourage use.
Please make new videos, post updated examples, etc.

**Comment by nokia6290** [comment]
Sounds like the serverpod is not ready for production

**Comment by juppdes** [comment]
have not even reacted to this topic, or they are too busy, or they are not giving due importance to what users think

**Comment by SandPod** [comment]
Hey @juppdes!

Sorry for the late reply on this one.
I think we tend to be quicker in our replies around more technical questions.

Short answer is YES!

We understand the documentation is dragging and some of our tutorials are outdated and are working on improving the situation.

From a documentation perspective, we are revising our processes around introducing and working on features to provide valid documentation faster. From a tutorial perspective, we are evaluating involving multiple technical writers to produce new and updated tutorials.

The effect of these changes might take time to materialize, but I want to stress that this is a high priority for Serverpod.

If a feature exists that users don't know how to use, it doesn't exist.

**Reply by dkbast** [reply]
A nitpick which I have been running into a couple of times is that I often end up in the wrong "version" of docs.serverpod.dev (usually because some information is there which is not available in the most recent version - figuring out if the information is still relevant is a bit tricky) - maybe there is a better way to archive the docs? Flutter has api docs for master & stable only and tutorials are usually only for "stable".

**Reply by SandPod** [reply]
I have noticed this issue as well. We are using docusaurus for our docs and with our configuration it doesn't replace old pages. This leads to links to old docs pages still being valid and serving that version of the documentation.

**Comment by jason-downs** [comment]
My 2 cents on this topic is that the Streming & Messaging section needs much more fleshing out with a lot more examples. I was eventually able to figure out how to do it in my app with the help of the Pixorama example, but ideally we should be able to have enough to work off of based on the docs alone and have the sample apps like Pixorama be "bonus" content that reinforces what is laid on in the docs. (It also did not help matters that the way Pixorama does streaming is somewhat different than what's in the docs). 

As for one example: for a relative newbie in this domain like myself, it was not clear at all what the difference was between `postMessage` and `sendStreamMessage` until I dug into the sample Pixorama code and parsed it out. This is something that may be obvious to people who have worked with streaming connections in other frameworks but was certainly confusing to me in the beginning, and something the docs address minimally if at all.

**Reply by KomosSolutions** [reply]
For anyone else needing some help with serverpod commands, or just remembering them after coming back to a serverpod project after a break, theres a Toolbox app worth looking into: https://github.com/KomosSolutions/serverpod_toolbox

**Reply by jason-downs** [reply]
Thanks @KomosSolutions this looks interesting, I will check it out! By the way, now with the recent revamp of the streaming functionality, my comment above might no longer apply; I'm not sure since i haven't used the new methods yet and haven't looked at the docs for it since the changes.

**Reply by SandPod** [reply]
@jason-downs - The new streaming interface should be well-documented. We spent a lot of time both making the interface as intuitive as possible, but also provide comprehensive documentation when the feature was launched. So far, most people seem to have found what they need.

I'm in the middle of updating the `Messaging` section as well (breaking it out into separate page).
